### ISSUE TITLE
Notifications dismiss themselves

### ISSUE THREAD

**Issue by chances** [staysail/nova-dvelop#8]
I use the [dlang.org install script](https://dlang.org/install.html) to install LDC. Nova have trouble finding my D environment. Regardless, this extension shows error notifications while trying to initialize the language server.

These notifications dismiss themselves after less than about 6 seconds. Notifications from Nova itself and other extensions let me read and dismiss them myself.

**Comment by gdamore** [comment]
Yeah, I think for error notifications like this we should do that.  I think part of this is how serve-d uses the LSP API to send these notifications -- I don't think they are classed as "errors" needing to be acknowledged.

I know that for other ones, like the "Language server restarted." notification, they are meant to be auto-dismiss, because there is no further user action needed.  But notices that need to be acknowledged or corrected should not automatically dismiss.

**Comment by gdamore** [comment]
@webfreak001 - any ideas here?

**Comment by WebFreak001** [comment]
these notifications are sent as error messages / have MessageType error - so probably a bug on your end? They stay there on vscode; unless of course it comes from restarting the language server, where it might clear the notifications from the previous run (vscode doesn't do that though, although there is no specification what to do)

**Comment by gdamore** [comment]
Errors are displayed using nova's`workspace.showErrorMessage` API and warnings are displayed using `workspace.showWarningMessage`. 

These don't provide any way to make the display modal. Nova claims it may bring the workspace to the foreground, but it says nothing about requiring confirmation, and there is no way to make them so using that API.

I think this is a limitation of Nova itself.

**Comment by gdamore** [comment]
I'm going to close this for now, because there is no way I can do anything more with it.  I'll also submit this to the Nova people.

**Comment by edwardloveall** [comment]
One option here is to use a [NotificationRequest](https://docs.nova.app/api-reference/notification-request/#constructoridentifier) instead of `workspace.showErrorMessage`. It's not documented well, but if you don't add `actions` the notification will auto-dismiss. But if you do add `actions` the alert will stay on the screen until dismissed.
